security/kubelogin-poc/README.md

```diff
@@ -84,13 +60,149 @@ the generated CA certificate in previous step. Using expired or non-trusted
 certificates causes apiserver not trust Dex, and login attempts will fail even
 if the Dex token kubectl gained is valid.
 
-See [dex.yaml](k8s/dex.yaml).
-
 We also need to add a line to `/etc/hosts` to create fake Dex domain:
-`127.0.0.1       dex.example.com` or simply change all occurences of
+`127.0.0.1       dex.example.com` or simply change all occurrences of
 `dex.example.com` with `127.0.0.1`. Here we use the `dex.example.com` in order
 to make a difference between Dex host and other services.
 
+Depending on the connector and grant type you want to configure, choose the
+appropriate configs below.
+
+#### Password connector
+
+Password flow is enabled only, if Dex config is set to allow it by setting one
+of the connectors as `passwordConnector`:
+
+```yaml
+oauth2:
+  passwordConnector: ldap
+```
+
+If this config is not present, password grant type is rejected as
+`not supported`. Most of the connectors do not support username/password.
+`local` and `ldap` do.
+
+#### Grant types
+
+Kubelogin and Dex support following grant types:
+
+- auto (automatic selection)
+- authcode (token passthrough via web browser callback)
+- authcode-keyboard (generates code you need to paste yourself)
+- device-code (like authcode, but for a device of certain code)
+- password (supply username and password on the command line, not enabled by default)
+
+Grant type is specified with `--grant-type=<type>` on command line during setup.
+
+##### Grant type: password
+
+<https://oauth.net/2/grant-types/password/>
+
+If we want to achieve headless CLI login, we need to use `password` grant type.
+It should be noted that password grants are removed in OAuth 2.1 spec, and kind
+of exist outside the OAuth2 spec. Password grants skip the auth code flow, and
+exchange credentials directly to tokens. For this reason, a client secret must
+be passed to user to be included in their kubeconfig and PKCE is not available.
```

Comment:
```suggestion
be passed to user to be included in their kubeconfig when PKCE is not available.
``` 
or
```suggestion
be passed to user to be included in their kubeconfig since in this case PKCE is not available.
```

Reply:
done